default.nix

```diff
@@ -0,0 +1,12 @@
+let
+  nix-pre-commit-hooks = import (builtins.fetchTarball "https://github.com/cachix/pre-commit-hooks.nix/tarball/master");
+in {
+  pre-commit-check = nix-pre-commit-hooks.run {
+    src = ./.;
```

Comment:
be aware that touching any file (even creating new file which is git ignored) within the directory will invalidate nix cache with this. This includes DS_Store on macos etc. It might or might not be a big problem but usually it's recommended to use some gitignore helper or some sub directory where temp files etc don't get created to avoid changes of inputs to nix.

Reply:
just adding gitignore won't be enough to make this work with git. You can check https://github.com/siers/nix-gitignore or look for similar projects. Anyway you can even just ignore this problem in first version but I think it's still worth noting because you might run into nix rebuilding a lot a be confused why this happens.

Reply:
Added, thank you! 😉